src/record_layout.zig

```diff
@@ -0,0 +1,459 @@
+const std = @import("std");
+const Type = @import("Type.zig");
+const Attribute = @import("Attribute.zig");
+const Compilation = @import("Compilation.zig");
+const Parser = @import("Parser.zig");
+const Record = Type.Record;
+const Field = Record.Field;
+const TypeLayout = Type.TypeLayout;
+
+// almost all the code for record layout
+// was liberally copied from this repro
+// https://github.com/mahkoh/repr-c
+// specifically the code in "repc/impl/src/" directory.
+// At the time of this writing (Q3 2022)
+// that code marked as "SPDX-License-Identifier: MIT OR Apache-2.0"
+// so is compatible with arocc's MIT licence
+
+const BITS_PER_BYTE: u29 = 8;
+
+const SysVContext = struct {
+    // Does the record has an __attribute__((packed)) annotation.
+    attr_packed: bool,
+    // The value of #pragma pack(N) at the type level if any.
+    max_field_align_bits: ?u64,
+    // The alignment of this record.
+    aligned_bits: u32,
+    is_union: bool,
+    // The size of the record. This might not be a multiple of 8 if the record contains bit-fields.
+    // For structs, this is also the offset of the first bit after the last field.
+    size_bits: u64,
+    fn init(ty: *Type, comp: *const Compilation, parser: ?*const Parser) SysVContext {
```

Comment:
```suggestion
    size_bits: u64,

    fn init(ty: *Type, comp: *const Compilation, parser: ?*const Parser) SysVContext {
```
There should be empty lines between functions and other functions/fields.